src/Types/JsonObjectType.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Types;
+
+/**
+ * Type generating json objects values
+ */
+class JsonObjectType extends JsonType
```

Comment:
Let's not introduce a type hierarchy just to reuse some code. You can move commonly used code into a trait that both types use.

I understand that we already have a hierarchy `JsonbType` to `JsonType`, but we should rather resolve that (in 5.0) instead of making the type hierarchy even weirder. Both of your new types should extend `Doctrine\DBAL\Types\Type`.